integration/ledger/snapshot_test.go

```diff
@@ -0,0 +1,1049 @@
+	By("getting current channel config from an existing peer")
+	existingPeers := n.PeersWithChannel(channelID)
+	Expect(len(existingPeers) > 0).Should(BeTrue())
+	currentConfig := nwo.GetConfig(n, existingPeers[0], orderer, channelID)
+	updatedConfig := proto.Clone(currentConfig).(*common.Config)
+
+	By("updating the channel config to include a new org")
+	sess, err := n.ConfigTxGen(commands.PrintOrg{
+		ConfigPath: n.RootDir,
+		PrintOrg:   org.Name,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(sess, n.EventuallyTimeout).Should(gexec.Exit(0))
+	orgGroup := &ordererext.DynamicOrdererOrgGroup{ConfigGroup: &common.ConfigGroup{}}
+	err = protolator.DeepUnmarshalJSON(bytes.NewBuffer(sess.Out.Contents()), orgGroup)
+	Expect(err).NotTo(HaveOccurred())
+	updatedConfig.ChannelGroup.Groups["Application"].Groups[org.Name] = orgGroup.ConfigGroup
+	nwo.UpdateConfig(n, orderer, channelID, currentConfig, updatedConfig, true, existingPeers[0], n.Peers...)
+}
+
+func joinBySnapshot(n *nwo.Network, orderer *nwo.Orderer, peer *nwo.Peer, channelID string, snapshotDir string, lastBlockInSnapshot int) {
+	channelHeight := nwo.GetMaxLedgerHeight(n, channelID, n.PeersWithChannel(channelID)...)
+
+	By(fmt.Sprintf("joining a peer via snapshot %s", snapshotDir))
+	n.JoinChannelBySnapshot(snapshotDir, orderer, peer)
+
+	By("calling JoinBySnapshotStatus")
+	checkStatus := func() []byte { return n.JoinBySnapshotStatus(orderer, peer) }
+	Eventually(checkStatus, n.EventuallyTimeout, 10*time.Second).Should(ContainSubstring("No joinbysnapshot operation is in progress"))
+
+	By("verifying blockchain info on peer " + peer.ID())
+	sess, err := n.PeerUserSession(peer, "Admin", commands.ChannelInfo{
+		ChannelID: channelID,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(sess, n.EventuallyTimeout).Should(gexec.Exit(0))
+	channelInfoStr := strings.TrimPrefix(string(sess.Buffer().Contents()[:]), "Blockchain info:")
+	var bcInfo = common.BlockchainInfo{}
+	err = json.Unmarshal([]byte(channelInfoStr), &bcInfo)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(bcInfo.Height).To(Equal(uint64(channelHeight)))
+	Expect(bcInfo.BootstrappingSnapshotInfo.LastBlockInSnapshot).To(Equal(uint64(lastBlockInSnapshot)))
+
+	By("waiting for all peers to have the same ledger height")
+	channelPeers := n.PeersWithChannel(channelID)
+	maxLedgerHeight := nwo.GetMaxLedgerHeight(n, channelID, channelPeers...)
+	nwo.WaitUntilEqualLedgerHeight(n, channelID, maxLedgerHeight, channelPeers...)
+}
+
+// verify the following functions for the peer after joining a channel by snapshot
+// - invoke/query chaincode
+// - upgrade chaincode
+// - channel config update
+// - join new channel
+// - qscc
+// - query history
+func verifyPeerAfterJoinBySnapshot(helper *marblesTestHelper, peer *nwo.Peer, channelID string, chaincode *nwo.Chaincode, newCCSequence string,
+	lastBlockInSnapshot int, oldTxid string, testKey string, availableHistoryResult []*marbleHistoryResult) []*marbleHistoryResult {
```

Comment:
parameters names need to be clearer. It's hard to see what some of them are in the context of "verifyPeerAfterJoinBySnapshot". See related comment above about using a struct for test data related params

Reply:
I split this function into multiple parts, some directly called from main flow and others in more specific helper functions.

Reply:
Have split this functioninto multiple parts, some directly called from main flow and others in more specific helper functions.